internal/cluster/registrar.go

```diff
@@ -118,7 +118,6 @@ func (r *registrar) DeregisterCluster(ctx context.Context, cluster *common.Clust
 	}
 
 	if txErr = common.DeleteRecordsByClusterID(tx, *cluster.ID, []interface{}{
-		&models.Host{},
```

Comment:
The code need to be also merged with the cluster cleaning code permanentClusterDeletion so both will be V2 and Unbound compatible

Reply:
Seems that Garbage collector is deleting the host in a different function: https://github.com/openshift/assisted-service/blob/master/internal/garbagecollector/garbageCollector.go#L85

I opened a task to update Garbage collector to delete the infraenv objects also: https://issues.redhat.com/browse/MGMT-8178